doc/tutorial.rst

```diff
@@ -560,8 +560,8 @@ Consider this example:
     #define lid(N) ((int) get_local_id(N))
     ...
       for (int i_outer = 0; i_outer <= -1 + (15 + n) / 16; ++i_outer)
-        for (int i_inner = 0; i_inner <= (-16 + n + -16 * i_outer >= 0 ? 15 : -1 + n + -16 * i_outer); ++i_inner)
-          a[16 * i_outer + i_inner] = 0.0f;
+        for (int i_inner = 0; i_inner <= ((char)(-16 + n + -16 * i_outer >= 0) ? 15 : -1 + n + -16 * i_outer); ++i_inner)
```

Comment:
Looking at them now, these `char` casts are a bit hard to bear. Any chance we can have type inference return the right bool-like type (maybe get it from the target) to avoid these casts being emitted?

Reply:
Sure thing. It would be correct to cast these to `bool` in CL so long as they aren't written to global arrays, right?

(And you wouldn't rather just omit the cast entirely? I wouldn't at all mind revising the type casting logic to allow for more specialization, it's just a matter of choosing how to do so.)

Reply:
> It would be correct to cast these to bool in CL so long as they aren't written to global arrays, right?

Reading/writing `GlobalArg`s and reading `ValueArg`s would be the main way in which `bool`s could find their way in. Are you saying you're thinking of adding hooks to all those paths?

> And you wouldn't rather just omit the cast entirely?

Of course! :)

Reply:
Ah, sorry, I misread your first comment here as "cast to bool instead of char." I'll give it (it being something that omits the casting) a stab soon.

Reply:
I addressed this by first adding the class attribute `use_int8_for_bool` to `TargetBase` (let me know if you'd prefer it just on `CFamilyTarget`) and adding it to `CFamilyTarget`'s hash and comparison fields. Type inference queries this to decide whether to return `int8` or `bool8` (replacing the prior `np.int32`, which seemed wrong?). Same modification for `ExpressionToCExpressionMapper.map_if`.

Reply:
(sorry for my repeated failure to type correctly, should be good now.)